src/instructlab/cli/data/generate.py

```diff
@@ -247,22 +263,60 @@ def generate(
         }
     )
 
+    if teacher_model_id:
+        try:
+            teacher_model_config = resolve_model_id(
+                teacher_model_id, ctx.obj.config.models
+            )
+            if not teacher_model_config:
+                raise ValueError(
+                    f"Teacher model with ID '{teacher_model_id}' not found in the configuration."
+                )
+            model_path = teacher_model_config.path
+        except ValueError as ve:
+            click.secho(f"failed to locate teacher model by ID: {ve}", fg="red")
+            raise click.exceptions.Exit(1)
+
     # determine student model arch from train section of config and pick system prompt to
     # pass to SDG appropriately
-    student_model_path = pathlib.Path(ctx.obj.config.train.model_path)
-    student_model_arch = get_model_arch(student_model_path)
-    system_prompt = get_sysprompt(student_model_arch)
+    system_prompt, legacy_pretraining_format = None, None
+    if not student_model_id:
+        student_model_path = pathlib.Path(ctx.obj.config.train.model_path)
```

Comment:
yeah this can be the default in the flag using `config_class`

Reply:
Can we please avoid this for now? I would like to leave it as something where it's merely a lookup for models in the config, rather than something with a known default that will be referenced in the existing system. This way we can keep the logic simple and change it later as needed.

Reply:
not quite - this default is a path, not a model id yet 
the way im imagining it, if we dont have to care about breaking changes as such we can rename all the model_path fields to either `model_id` or just `model` and have them reference model ids instead of paths. Until we do that, setting defaults in the click flags wont achieve much